build/RepositoryBuild.targets

```diff
@@ -23,13 +24,22 @@
   </Target>
 
   <Target Name="_BuildRepositories" DependsOnTargets="GetRepoBatches" Condition=" @(RepositoryBuildOrder->Count()) != 0 ">
+    <RemoveDir Directories="$(_SignRequestIntermediateOutputPath)" />
+    <MakeDir Directories="$(_SignRequestIntermediateOutputPath)" />
+
     <MSBuild
       Projects="@(BatchedRepository)"
       BuildInParallel="true"
       StopOnFirstFailure="true"
       Targets="_BuildRepository"
       Properties="BuildGroup=%(BatchedRepository.BuildGroup);BuildNumber=$(BuildNumber);IsFinalBuild=$(IsFinalBuild);Configuration=$(Configuration)" />
 
+    <ItemGroup>
+      <SignRequests Include="$(_SignRequestIntermediateOutputPath)*.xml" />
+    </ItemGroup>
+
+    <MergeXmlFiles Files="@(SignRequests)" OutputPath="$(ArtifactsDir)signrequest.xml" />
```

Comment:
Neat! Is doing this faster than sending multiple sign requests?

Reply:
Yes. There is a fixed cost each time you have to round-trip to code-signing which is pretty high. So the more files you can send per batch, the better.